tools/model_tools/src/openvino/model_zoo/quantize_engine/quantizer.py

```diff
@@ -0,0 +1,195 @@
+import json
+import os
+import shutil
+import sys
+import tempfile
+import yaml
+
+from pathlib import Path
+from typing import Set, List
+
+from openvino.model_zoo import _common
+
+
+DEFAULT_POT_CONFIG_BASE = {
+    'compression': {
+        'algorithms': [
+            {
+                'name': 'DefaultQuantization',
+                'params': {
+                    'preset': 'performance',
+                    'stat_subset_size': 300,
+                },
+            },
+        ],
+    },
+}
+
+
+class Quantizer:
+    def __init__(self, python: str, requested_precisions: str, output_dir: Path, model_dir: Path, pot: Path,
+                 dataset_dir: Path, dry_run: bool):
+        self.python = python
+        self.pot_cmd_prefix = pot
+        self.requested_precisions = requested_precisions
+        self.output_dir = output_dir or model_dir
+        self.model_dir = model_dir
+        self.dry_run = dry_run
+        self.dataset_dir = dataset_dir
+
+    @property
+    def dataset_dir(self) -> Set[str]:
+        return self._dataset_dir
+
+    @dataset_dir.setter
+    def dataset_dir(self, value: str = None):
+        # We can't mark it as required, because it's not required when --print_all is specified.
+        # So we have to check it manually.
+        if not value:
+            sys.exit('--dataset_dir must be specified.')
+        self._dataset_dir = value
+
+    @property
+    def requested_precisions(self) -> Set[str]:
+        return self._requested_precisions
+
+    @requested_precisions.setter
+    def requested_precisions(self, value: Set[str] = None):
+        unknown_precisions = value - _common.KNOWN_QUANTIZED_PRECISIONS.keys()
+        if unknown_precisions:
+            sys.exit('Unknown precisions specified: {}.'.format(', '.join(sorted(unknown_precisions))))
+
+        self._requested_precisions = value
+
+    @property
+    def pot_cmd_prefix(self) -> Path:
+        return self._pot_cmd_prefix
+
+    @pot_cmd_prefix.setter
+    def pot_cmd_prefix(self, value: str = None):
```

Comment:
for me from API user perspective this place looks slightly confusing. The class assumes that pot path already predefined explicitly or required to set explicitly to None to trigger executable check behavior. Maybe no need to use setter here and explicitly use method for setting?